internal/controller/redis_connector_test.go

```diff
@@ -0,0 +1,280 @@
+			connIndex: []string{"01:node-a:" + testHost, "01:node-b:" + testHost},
+			podIndex:  []string{"01:node-a", "01:node-b"},
+			err:       nil,
+		},
+		{
+			account:   "02",
+			nodeID:    "node-a",
+			hostname:  testHost,
+			accIndex:  []string{"node-a:" + testHost},
+			connIndex: []string{"01:node-a:" + testHost, "01:node-b:" + testHost, "02:node-a:" + testHost},
+			podIndex:  []string{"01:node-a", "01:node-b", "02:node-a"},
+			err:       nil,
+		},
+	}
+
+	for _, tc := range tests {
+		got := RegisterWithRedis(c, tc.account, tc.nodeID, tc.hostname)
+		if got != tc.err {
+			t.Fatalf("expected: %v, got: %v", tc.err, got)
+		}
+		assert.Equal(t, c.Get(tc.account+":"+tc.nodeID).Val(), tc.hostname)
+		assert.Equal(t, c.SMembers(tc.account).Val(), tc.accIndex)
+		assert.Equal(t, c.SMembers("connections").Val(), tc.connIndex)
+		assert.Equal(t, c.SMembers(tc.hostname).Val(), tc.podIndex)
+	}
+}
+
+func TestDuplicateRegisterWithRedis(t *testing.T) {
+	s, _ := miniredis.Run()
+	defer s.Close()
+
+	c := newTestRedisClient(s.Addr())
+
+	tests := []struct {
+		account      string
+		nodeID       string
+		hostname     string
+		expectedHost string
+		accIndex     []string
+		connIndex    []string
+		podIndex     []string
+		err          error
+	}{
+		{
+			account:      "01",
+			nodeID:       "node-a",
+			hostname:     testHost,
+			expectedHost: testHost,
+			accIndex:     []string{"node-a:" + testHost},
+			connIndex:    []string{"01:node-a:" + testHost},
+			podIndex:     []string{"01:node-a"},
+			err:          nil,
+		},
+		{
+			account:      "01",
+			nodeID:       "node-a",
+			hostname:     "dupe-conn",
+			expectedHost: testHost,
+			accIndex:     []string{"node-a:" + testHost},
+			connIndex:    []string{"01:node-a:" + testHost},
+			podIndex:     []string{},
+			err:          DuplicateConnectionError{},
+		},
+		{
+			account:      "02",
+			nodeID:       "node-a",
+			hostname:     testHost,
+			expectedHost: testHost,
+			accIndex:     []string{"node-a:" + testHost},
+			connIndex:    []string{"01:node-a:" + testHost, "02:node-a:" + testHost},
+			podIndex:     []string{"01:node-a", "02:node-a"},
+			err:          nil,
+		},
+		{
+			account:      "02",
+			nodeID:       "node-a",
+			hostname:     "dupe-conn",
+			expectedHost: testHost,
+			accIndex:     []string{"node-a:" + testHost},
+			connIndex:    []string{"01:node-a:" + testHost, "02:node-a:" + testHost},
+			podIndex:     []string{},
+			err:          DuplicateConnectionError{},
+		},
+	}
+
+	for _, tc := range tests {
+		got := RegisterWithRedis(c, tc.account, tc.nodeID, tc.hostname)
+		if got != tc.err {
+			t.Fatalf("expected: %v, got: %v", tc.err, got)
+		}
+		assert.Equal(t, c.Get(tc.account+":"+tc.nodeID).Val(), tc.expectedHost)
+		assert.Equal(t, c.SMembers(tc.account).Val(), tc.accIndex)
+		assert.Equal(t, c.SMembers("connections").Val(), tc.connIndex)
+		assert.Equal(t, c.SMembers(tc.hostname).Val(), tc.podIndex)
```

Comment:
This tests that a new pod entry is not made for the duplicate connection.  This is certainly needed, but I think we should also make sure that the original pod entry has not been damaged.  What do you think?

Reply:
By the original pod entry do you mean the state of the pod index? If so I can add an additional check to make sure that the index for the other pod wasn't modified as part of this registration.

Reply:
Yep